src/deploy/functions/validate.ts

```diff
@@ -15,11 +15,12 @@ const cjson = require("cjson");
 
 /**
  * Check that functions directory exists.
- * @param options options object.
+ * @param options options object. In prod is an args.Options; in tests can just be {cwd: string}
  * @param sourceDirName Relative path to source directory.
  * @throws { FirebaseError } Functions directory must exist.
  */
-export function functionsDirectoryExists(options: object, sourceDirName: string): void {
+export function functionsDirectoryExists(options: { cwd: string }, sourceDirName: string): void {
```

Comment:
Should this be `options: { cwd?: string, configPath?: string}`, like what resolveProjectPath takes? Its not immediately clear to me if this will only ever have cwd, or if it will have configPath depending on the options passed in.

Reply:
I'm a bit confused. Are you saying that `sourceDirName` should be part of the passed options? This is always an `args.Options`, which always has a `cwd`. I just changed it from `args.Options` because tests were annoying to fix if we required a full `args.Options`.

Reply:
Not quite - functionsDirectoryExists calls [resolveProjectPath](https://github.com/firebase/firebase-tools/blob/master/src/projectPath.ts#L11), which has the following signature: ```export function resolveProjectPath(
  options: { cwd?: string; configPath?: string },
  filePath: string
):``` 
I'm wondering if theres any edge cases where options.configPath would be defined here - from https://github.com/firebase/firebase-tools/blob/master/src/command.ts#L237, it looks like it comes from the --config flag.

Reply:
Documenting the configPath option. Making it optional since the existing tests don't use it, though it should always be present per my research into the Options type. Since both prod and test provide a CWD I'm leaving that as required.